examples/apps/tree-comparison/src/model/treeInventoryList.ts

```diff
@@ -0,0 +1,188 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	AllowedUpdateType,
+	ForestType,
+	typeboxValidator,
+	TypedTreeChannel,
+	TypedTreeFactory,
+} from "@fluid-experimental/tree2";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { v4 as uuid } from "uuid";
+
+import type { IInventoryItem, IInventoryList } from "../modelInterfaces";
+import { InventoryItem } from "./inventoryItem";
+import { InventoryNode, InventoryField, InventoryItemNode, schema } from "./schema";
+
+const factory = new TypedTreeFactory({
+	// REV: I'm not exactly sure why a validator should be passed here?  Like what it's used for,
+	// so it's hard to know what a "correct" choice would be as a result.
+	jsonValidator: typeboxValidator,
```

Comment:
Doc comment here:
	
       /**
	 * JSON Validator which should be used to validated persisted data SharedTree handles.
	 * See {@link noopValidator} and {@link typeboxValidator} for out-of-the-box implementations.
	 *
	 * SharedTree users are encouraged to use a non-trivial validator (i.e. not `noopValidator`) whenever
	 * reasonable: it gives better fail-fast behavior when unexpected encoded data is found, which reduces
	 * the risk of further data corruption.
	 */

To be fair, that's missing the negatives and why it's not just auto-included: using a nontrivial validator comes with smallish but noticeable runtime overhead (10-20%) and bundle size (varies based on the one you use, but if your app already uses a JSON schema validator you can re-use that and avoid paying such a bundling cost)

Reply:
I saw that but the problem is I don't know what it's validating - like what does "validate persisted data SharedTree handles" mean.  Is it validating the snapshot, the object-type payloads in the tree, serialized IFluidHandles stored in the tree, etc.  Is it JSON that I'm providing in some way (like in my stored data I would guess?) or is it JSON that is out of my control (e.g. some wire format used by SharedTree)?

Reply:
#17768